Mathlib/Tactic/Linter/UnusedSetOptionIn.lean

```diff
@@ -0,0 +1,65 @@
+/-
+Copyright (c) 2024 Damiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damiano Testa
+-/
+
+import Mathlib.Tactic.Lemma
+import Lean.Elab.Command
+import Lean.Linter.Util
+
+/-!
+The `unusedSetOptionIn` linter warns against "most external" unused `set_option ... in`s.
+-/
+
+open Lean Elab Command
+
+/-- converts
+* `theorem x ...` to  `some (example ... , x)`,
+* `lemma x ...`   to  `some (example ... , x)`,
+* `example ...`   to ``some (example ... , `example)``,
```

Comment:
Is this correct?
```suggestion
* `example ...`   to ``some (example ... , 'example')`,

Reply:
I ended up going for your version: the reason behind what I had, was that I was intepreting the `example` as a `Name`, but it does not matter much either way -- it is a syntax node anyway!